core/src/main/java/org/bitcoinj/wallet/DeterministicKeyChain.java

```diff
@@ -332,6 +332,31 @@ public DeterministicKeyChain(DeterministicKey watchingKey) {
         initializeHierarchyUnencrypted(watchingKey);
     }
 
+    /**
+     * Creates a deterministic key chain from a watched or spendable account key.  If  <code>isWatching</code> flag is set,
+     * then creates a deterministic key chain that watches the given (public only) root key.  You can use this to calculate
+     * balances and generally follow along, but spending is not possible with such a chain.  If it is not set, then this
+     * creates a deterministic key chain that allows spending. If <code>isFollowing</code> flag is set(only allowed
+     * if <code>isWatching</code> is set) then this keychain follows some other keychain.  In a married wallet following
+     * keychain represents "spouse's" keychain.
+     */
+    public DeterministicKeyChain(DeterministicKey key, boolean isFollowing, boolean isWatching) {
+        if(isWatching)
+            checkArgument(key.isPubKeyOnly(), "Private subtrees not currently supported for watching keys: if you got this key from DKC.getWatchingKey() then use .dropPrivate().dropParent() on it first.");
+        else
+            checkArgument(key.hasPrivKey(), "Private subtrees are required.");
+        checkArgument(isWatching ? true : !isFollowing, "Can only follow a key that is watched");
```

Comment:
¿Would this not be the same as `checkArgument(isWatching || !isFollowing, ...`?

Reply:
Indeed :smile: